Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -90,165 +83,1086 @@ function Sync-AzSOfflineMarketplaceItem {
                     })
             }
 
+            'solution' {
+                Write-output ([pscustomobject]@{
+                        Id          = $product.name.Split('/')[-1]
+                        Type        = "Solution"
+                        Name        = $product.properties.displayName
+                        Description = $product.properties.description
+                        Publisher   = $product.properties.publisherDisplayName
+                        Version     = $product.properties.productProperties.version
+                        Size        = Set-String -size $product.properties.payloadLength
+                    })
+            }
+
+            'resourceProvider' {
+                Write-output ([pscustomobject]@{
+                        Id          = $product.name.Split('/')[-1]
+                        Type        = "Resource Provider"
+                        Name        = $product.properties.displayName
+                        Description = $product.properties.description
+                        Publisher   = $product.properties.publisherDisplayName
+                        Version     = $product.properties.productProperties.version
+                        Size        = Set-String -size $product.properties.payloadLength
```

Comment:
the Set-String function looks a bit odd. Instead of using those magic numbers, powershell has "1GB" and "1MB" constants you can divide by. Please consider renaming this function as "Set-String" is terrible. Perhaps "Get-SizeDisplayString" or something like that.